plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightClientManager.java

```diff
@@ -128,13 +141,16 @@ private void buildClientAndAddToPool(Location location, DiscoveryNode node) {
             );
             return;
         }
-        flightClients.computeIfAbsent(node.getId(), key -> buildClient(location));
+        if (closed) {
+            return;
+        }
+        FlightClient flightClient = buildClient(location);
+        flightClients.put(node.getId(), new ClientHolder(location, flightClient));
```

Comment:
Let's avoid creating client instances in case of races:
```suggestion
 flightClients.computeIfAbsent(node.getId(), 
            nodeId -> new ClientHolder(location, buildClient(location)));
```